tsdat/io/base/archive_reader.py

```diff
@@ -5,6 +5,8 @@
 from .data_reader import DataReader
 
 
+# TODO: Generally it's bad practice to not implement all abstract methods from an imported base
+#  class. The DataReader abstract class has the `read` method, which is not implemented here.
```

Comment:
Agreed. The two classes that inherit from this have nearly identical `read` methods, save for how the archive is unpacked. Maybe we can have the base `ArchiveReader` class define an `unpack()` abstract method and an implementation of the `read()` method. Then the child classes (`ZipReader`, `TarReader`) can just implement `unpack()`)

Reply:
If I'm understanding correctly, `ArchiveReader` is, itself an abstract class? If that's the case then we just need to alter its class init to be `class ArchiveReader(DataReader, ABC):` and we're good.

Reply:
Yeah, `ArchiveReader` is an abstract class; it should inherit from `ABC`

Reply:
Resolved in 7ec5abc5a42f7d5e565f562321c664aeb5c46bcd.